setup.py

```diff
@@ -608,17 +649,21 @@ def get_common_options(build_ext):
         raise DistutilsError('HOROVOD_GPU_BROADCAST=%s is invalid, supported '
                              'values are "", "MPI", "NCCL".' % gpu_broadcast)
 
+    have_cuda = False
+    have_rocm = False
+    gpu_include_dirs = gpu_lib_dirs = gpu_macros = []
     if gpu_allreduce or gpu_allgather or gpu_broadcast:
-        have_cuda = True
-        cuda_include_dirs, cuda_lib_dirs = get_cuda_dirs(build_ext, cpp_flags)
-    else:
-        have_cuda = False
-        cuda_include_dirs = cuda_lib_dirs = []
+        if os.environ.get('HOROVOD_GPU_ROCM'):
```

Comment:
Can we get rid of this environment variable and ask the user to set `HOROVOD_GPU_ALLREDUCE=RCCL` instead?

Reply:
I think at best we could use `HOROVOD_GPU_ALLREDUCE=RCCL` to automatically set a missing `HOROVOD_GPU_ROCM`, but shouldn't we still support builds where RCCL is not used but we still build horovod for ROCm and assume a GPUDirect-capable MPI install?  Is this still a use case for CUDA?

Reply:
Correct, now we can use cuda-aware MPI without NCCL if we say (HOROVOD_GPU_ALLREDUCE=MPI or HOROVOD_GPU_BROADCAST=MPI).

Reply:
That's a good point.  How about we change it to `HOROVOD_GPU=ROCM`.  That way we leave open the possibility of other APIs as well, and potentially to auto-infer if not set.

Reply:
I'm assuming you would like the current behavior to remain the same, i.e., assume that CUDA is selected.  So something like the following?

```python
    have_cuda = False
    have_rocm = False
    gpu_include_dirs = gpu_lib_dirs = gpu_macros = []
    if gpu_allreduce or gpu_allgather or gpu_broadcast:
        gpu_type = os.environ.get('HOROVOD_GPU', 'CUDA')
        if gpu_type == 'CUDA':
            have_cuda = True
            gpu_include_dirs, gpu_lib_dirs = get_cuda_dirs(build_ext, cpp_flags)
        elif gpu_type == 'ROCM':
            have_rocm = True
            gpu_include_dirs, gpu_lib_dirs, gpu_macros = get_rocm_dirs(build_ext, cpp_flags)
        else:
            raise DistutilsError("Unknown HOROVOD_GPU type '%s'" % gpu_type)
```

Reply:
Yes, that's exactly what I had in mind.  Does that sound good to you?

Reply:
Done.